db/dcrpg/pgblockchain.go

```diff
@@ -5399,16 +5409,44 @@ func (pgb *ChainDB) GetExplorerTx(txid string) *exptypes.TxInfo {
 		log.Errorf("Cannot create MsgTx for tx %v: %v", txhash, err)
 		return nil
 	}
+
+	var isMix bool
+	var mixDenom int64
+	var mixCount uint32
+	if !txhelpers.IsStakeTx(msgTx) {
+		isMix, mixDenom, mixCount = txhelpers.IsMixTx(msgTx)
+		if !isMix {
+			var ticketPrice int64
+			if txraw.Confirmations > 0 {
+				ticketPrice = pgb.GetSBitsByHash(txraw.BlockHash)
+			} else {
+				sdiffRes, err := pgb.Client.GetStakeDifficulty()
+				if err == nil {
+					sdiff, _ := dcrutil.NewAmount(sdiffRes.CurrentStakeDifficulty) // err means sdiff==0
+					ticketPrice = int64(sdiff)
+				}
+			}
+			log.Infof("ticket price at %s: %d", txraw.BlockHash, ticketPrice)
+			isMix, mixCount, _ = txhelpers.IsMixedSplitTx(msgTx, ticketPrice)
+			if isMix {
+				mixDenom = ticketPrice
```

Comment:
The mixed output amounts when ticket buying is not just the current ticket price, but also includes extra DCR in order that the ticket purchase itself may consume just that output and still pay the required transaction fee.

Reply:
Thanks for pointing that out.  It needs to be `ticketOutAmt` as computed in `IsMixedSplitTx`: https://github.com/decred/dcrdata/pull/1610/files/fab622a77e4da638b8d88d8923e8486f490ae9ba#diff-74e3b548913f559d292b3e67d1cc1d47R92

Reply:
BTW, all the stolen code regarding this computation does not sit well with me for a number of reasons.  I'd like a more reliable way to do that.